sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemConfig.java

```diff
@@ -0,0 +1,111 @@
+        logOptions = new HttpLogOptions().setLogLevel(
+            (HttpLogDetailLevel) config.get(AzureFileSystem.AZURE_STORAGE_HTTP_LOG_DETAIL_LEVEL));
+        retryOptions = new RequestRetryOptions(
+            (RetryPolicyType) config.get(AzureFileSystem.AZURE_STORAGE_RETRY_POLICY_TYPE),
+            (Integer) config.get(AzureFileSystem.AZURE_STORAGE_MAX_TRIES),
+            (Integer) config.get(AzureFileSystem.AZURE_STORAGE_TRY_TIMEOUT),
+            (Long) config.get(AzureFileSystem.AZURE_STORAGE_RETRY_DELAY_IN_MS),
+            (Long) config.get(AzureFileSystem.AZURE_STORAGE_MAX_RETRY_DELAY_IN_MS),
+            (String) config.get(AzureFileSystem.AZURE_STORAGE_SECONDARY_HOST));
+        httpClient = (HttpClient) config.get(AzureFileSystem.AZURE_STORAGE_HTTP_CLIENT);
+
+        HttpPipelinePolicy[] policyArray = (HttpPipelinePolicy[]) config.get(AzureFileSystem.AZURE_STORAGE_HTTP_POLICIES);
+        Collections.addAll(policyList, policyArray != null ? policyArray : new HttpPipelinePolicy[0]);
+
+        blockSize = (Long) config.get(AzureFileSystem.AZURE_STORAGE_UPLOAD_BLOCK_SIZE);
+        putBlobThreshold = (Long) config.get(AzureFileSystem.AZURE_STORAGE_PUT_BLOB_THRESHOLD);
+        maxConcurrencyPerRequest = (Integer) config.get(AzureFileSystem.AZURE_STORAGE_MAX_CONCURRENCY_PER_REQUEST);
+        downloadResumeRetries = (Integer) config.get(AzureFileSystem.AZURE_STORAGE_DOWNLOAD_RESUME_RETRIES);
+
+        String fileStores = (String) config.get(AzureFileSystem.AZURE_STORAGE_FILE_STORES);
+        Collections.addAll(fileStoreNames, (fileStores != null ? fileStores : "").split(","));
+
+        skipInitialContainerCheck = (Boolean) config.get(AzureFileSystem.AZURE_STORAGE_SKIP_INITIAL_CONTAINER_CHECK);
+    }
+
+    public AzureFileSystemConfig(Configuration config) {
+        if (!CoreUtils.isNullOrEmpty(config.get(ENVIRONMENT_DEFAULT_ACCOUNT_NAME)) &&
+            !CoreUtils.isNullOrEmpty(config.get(ENVIRONMENT_DEFAULT_ACCOUNT_KEY))) {
+            sharedKeyCredential = new StorageSharedKeyCredential(
+                config.get(ENVIRONMENT_DEFAULT_ACCOUNT_NAME), config.get(ENVIRONMENT_DEFAULT_ACCOUNT_KEY));
+        } else {
+            sharedKeyCredential = null;
+        }
+        sasCredential = getConfigFromStringConfiguration(config, ENVIRONMENT_DEFAULT_SAS_TOKEN,
+            AzureSasCredential::new);
+
+        // HttpLogOptions constructor does the environment read for us
+        logOptions = new HttpLogOptions();
+        retryOptions = RequestRetryOptions.fromConfiguration(config,
+            config.get(ENVIRONMENT_DEFAULT_BLOB_ENDPOINT_SECONDARY));
+        httpClient = null; // cannot load from environment
+
+        // cannot load policies from environment
+
+        blockSize = getConfigFromStringConfiguration(config, ENVIRONMENT_DEFAULT_BLOCK_SIZE, Long::valueOf);
+        putBlobThreshold = getConfigFromStringConfiguration(config, ENVIRONMENT_DEFAULT_PUT_BLOB_THRESHOLD,
```

Comment:
Looks like these are slightly different names than what goes in the config map. Be sure to update the documentation on AzureFileSystemProvider and other places (readme, etc.) that detail how you can initialize a provider

Reply:
Yes that is still on the todo list. Unfortunately, the existingconfig map keys don't match expected naming conventions of environment variables. Will have to provide a new list as well as exactly what they map onto.